mat-process-api/V1/Validators/PostInitialProcessDocumentRequestValidator.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentValidation;
+using mat_process_api.V1.Boundary;
+
+namespace mat_process_api.V1.Validators
+{
+    public class PostInitialProcessDocumentRequestValidator : AbstractValidator<PostInitialProcessDocumentRequest>
+    {
+        public PostInitialProcessDocumentRequestValidator()
+        {
+            ValidatorOptions.CascadeMode = CascadeMode.StopOnFirstFailure;
+
+            RuleFor(postReq => postReq.processRef).NotNull().WithMessage("Process reference must be provided.").NotEmpty().WithMessage("Process reference must be provided.").Must(ValidateGuid)
+               .WithMessage("You need to provide a valid process reference.");
+            RuleFor(postReq => postReq.processType).NotNull().WithMessage("Process type must be provided.")
+                .DependentRules(() =>
+                {
+                    RuleFor(postReq => postReq.processType.value).NotEmpty().WithMessage("Process type value must be provided.")
+                     .NotEmpty().WithMessage("Process type value must be provided."); ;
```

Comment:
Duplicate Line, double semicolon. Might need to change one of them into "NotNull". Also, I think the WithMessage can be trimmed into a single call after both validations.

Reply:
Fixed that. Though apparently the 'null' check might be pointless. Unless someone decides to make it into 'int?' in the future.